<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <include 
        file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="world_name"  value="$(find vitarana_drone)/worlds/obstacles.world"/>
    </include>

	<node pkg="vitarana_drone" type="attitude_controller.py" name="attitude_controller" output="screen"/>

	<node pkg="vitarana_drone" type="position_controller.py" name="position_controller" output="screen">
        <param name="use_ff" value="False" /> <!-- "True", "False"-->
        <param name="use_ca" value="True" /> <!-- "True", "False"-->
    </node>

	<node pkg="vitarana_drone" type="motor_api.py" name="motor_api" output="screen"/>

    <node pkg="vitarana_drone" type="plotter.py" name="plotter" output="screen"/>

	<param name="robot_description"
           command="$(find xacro)/xacro '$(find vitarana_drone)/models/car/urdf/my_robot.xacro'" />

    <node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" output="screen"
          args="-param robot_description
                -urdf
                -model my_robot
                -x 10 -y 0 -z 0.1" />

    <node pkg="vitarana_drone" type="car_trajectory.py" name="car_trajectory" output="screen">
        <param name="trajectory" value="circle" />
        <param name="linear_velocity" value="0.0" />
        <param name="angular_velocity" value="0.0" />
    </node>

</launch>